torch_np/_wrapper.py

```diff
@@ -314,20 +313,98 @@ def identity(n, dtype=None, *, like=None):
 ###### misc/unordered
 
 
-@asarray_replacer()
+@_decorators.dtype_to_torch
 def corrcoef(x, y=None, rowvar=True, bias=NoValue, ddof=NoValue, *, dtype=None):
     if bias is not None or ddof is not None:
         # deprecated in NumPy
         raise NotImplementedError
+
+    # https://github.com/numpy/numpy/blob/v1.24.0/numpy/lib/function_base.py#L2636
     if y is not None:
-        # go figure what it means, XXX
-        raise NotImplementedError
+        x = array(x, ndmin=2)
+        if not rowvar and x.shape[0] != 1:
+            x = x.T
```

Comment:
In master you get:
```
>>> torch.randn(3).T
<stdin>:1: UserWarning: The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated and it will throw an error in a future release. Consider `x.mT` to transpose batches of matrices or `x.permute(*torch.arange(x.ndim - 1, -1, -1))` to reverse the dimensions of a tensor. (Triggered internally at /home/lezcano/git/pytorch/pytorch/aten/src/ATen/native/TensorShape.cpp:3571.)
tensor([-0.7610, -0.8684, -1.1861])
```
so better add a `y.ndim == 2`